homeassistant/components/advantage_air/climate.py

```diff
@@ -84,39 +84,29 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class AdvantageAirClimateEntity(AdvantageAirEntity, ClimateEntity):
     """AdvantageAir Climate class."""
 
-    @property
-    def temperature_unit(self):
-        """Return the temperature unit."""
-        return TEMP_CELSIUS
-
-    @property
-    def target_temperature_step(self):
-        """Return the supported temperature step."""
-        return PRECISION_WHOLE
-
-    @property
-    def max_temp(self):
-        """Return the maximum supported temperature."""
-        return 32
-
-    @property
-    def min_temp(self):
-        """Return the minimum supported temperature."""
-        return 16
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_max_temp = 32
+    _attr_min_temp = 16
 
 
 class AdvantageAirAC(AdvantageAirClimateEntity):
     """AdvantageAir AC unit."""
 
-    @property
-    def name(self):
-        """Return the name."""
-        return self._ac["name"]
-
-    @property
-    def unique_id(self):
-        """Return a unique id."""
-        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}'
+    _attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+    _attr_fan_modes = [FAN_AUTO, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+
+    def __init__(self, instance, ac_key):
+        """Initialize an AdvantageAir AC unit."""
+        super().__init__(instance, ac_key)
+        self._attr_name = self._ac["name"]
+        self._attr_unique_id = f'{self.coordinator.data["system"]["rid"]}-{ac_key}'
+        self._attr_hvac_modes = AC_HVAC_MODES
+        self._attr_hvac_modes = (
+            AC_HVAC_MODES + [HVAC_MODE_AUTO]
+            if self._ac.get("myAutoModeEnabled")
+            else None
+        )
```

Comment:
Wont this set have_modes to None if myAutoModeEnabled is true?
```suggestion
        self._attr_hvac_modes = AC_HVAC_MODES
        self._attr_hvac_modes += (
            [HVAC_MODE_AUTO]
            if self._ac.get("myAutoModeEnabled")
            else None
        )
```